src/main/scala/com/ing/wbaa/airlock/proxy/api/PostRequestActions.scala

```diff
@@ -16,22 +19,37 @@ trait PostRequestActions {
 
   protected[this] def emitEvent(s3Request: S3Request, method: HttpMethod, principalId: String): Future[Done]
 
-  def createAtlasLineage(response: HttpResponse, httpRequest: HttpRequest, userSTS: User, clientIPAddress: RemoteAddress): Unit =
+  protected[this] def setBucketPolicy(bucketName: String): Future[Unit]
+
+  private[this] def createAtlasLineage(response: HttpResponse, httpRequest: HttpRequest, userSTS: User, clientIPAddress: RemoteAddress): Unit =
     if (atlasSettings.atlasEnabled && (response.status == StatusCodes.OK || response.status == StatusCodes.NoContent))
       // delete on AWS response 204
       createLineageFromRequest(httpRequest, userSTS, clientIPAddress)
 
-  protected[this] def createBucketNotification(response: HttpResponse, httpRequest: HttpRequest, s3Request: S3Request,
+  private[this] def createBucketNotification(response: HttpResponse, httpRequest: HttpRequest, s3Request: S3Request,
       userSTS: User): Future[Done] =
     httpRequest.method match {
       case HttpMethods.POST | HttpMethods.PUT | HttpMethods.DELETE if kafkaSettings.kafkaEnabled && (response.status == StatusCodes.OK || response.status == StatusCodes.NoContent) =>
         emitEvent(s3Request, httpRequest.method, userSTS.userName.value)
       case _ => Future.successful(Done)
     }
 
+  private[this] def updateBucketPolicy(httpRequest: HttpRequest, s3Request: S3Request): Unit = {
+    lazy val matches = bucketRegex findFirstIn httpRequest.uri.path.toString()
```

Comment:
I would suggest to keep extraction of request values in as few places as possible. Currently we have S3Request that is beeing generated in ProxyDirectives. Maybe we should add bucket object in S3Request and populate it once?